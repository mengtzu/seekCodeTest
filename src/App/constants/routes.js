const root = '/';
const adPacks = '/adPacks';
const checkout = '/checkout';
const confirmation = '/confirmation';
const logout = '/Login/Logout';

export const routes = {
    root,
    adPacks,
    checkout,
    confirmation,
    logout
};

export default routes;
