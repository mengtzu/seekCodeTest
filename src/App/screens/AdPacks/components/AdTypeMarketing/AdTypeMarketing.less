@import (reference) "~seek-style-guide/theme";

.root {
    background-position: left top;
    background-size: 200px auto;
    background-repeat: no-repeat;
    padding-left: 220px;
    min-height: 205px;
}

.classic {
    &:extend(.root);

    background-image: url('./classic.png');
}

.standout {
    &:extend(.root);

    background-image: url('./standout.png');
}

.premium {
    &:extend(.root);

    background-image: url('./premium.png');
}