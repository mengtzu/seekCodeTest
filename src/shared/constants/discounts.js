export const DISCOUNT_TYPE_FLAT_RATE = 'flatRate';
export const DISCOUNT_TYPE_THRESHOLD = 'threshold';
export const DISCOUNT_TYPE_X_FOR_Y = 'xfory';
