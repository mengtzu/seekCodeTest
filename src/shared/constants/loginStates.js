export const AUTH_STATUS_UNAUTHENTICATED = 'pending'; //'unauthenticated' exposes the real SEEK login, don't want that!
export const AUTH_STATUS_AUTHENTICATED = 'authenticated';
