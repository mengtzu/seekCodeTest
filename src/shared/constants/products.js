export const CLASSIC = 'classic';
export const STANDOUT = 'standout';
export const PREMIUM = 'premium';
