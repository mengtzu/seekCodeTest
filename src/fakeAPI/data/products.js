import { CLASSIC, STANDOUT, PREMIUM } from '../../shared/constants/products';

let productsData = {};

//Bracket notation isn't necessarily preferred but here it lets us avoid having to iterate over this object in various places

productsData[CLASSIC] = {
    id: CLASSIC,
    basePrice: 269.99
};

productsData[STANDOUT] = {
    id: STANDOUT,
    basePrice: 322.99
};

productsData[PREMIUM] = {
    id: PREMIUM,
    basePrice: 394.99
}

export default productsData;