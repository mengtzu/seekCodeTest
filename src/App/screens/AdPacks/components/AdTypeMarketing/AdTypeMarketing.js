import React from 'react';
import styles from './AdTypeMarketing.less';
import { Text } from 'seek-style-guide/react';

import { CLASSIC, STANDOUT, PREMIUM } from '../../../../../shared/constants/products';

const getClassName = (adType) => {
    switch(adType) {
        case CLASSIC:
            return styles.classic;
            break;
        case STANDOUT:
            return styles.standout;
            break;
        case PREMIUM:
            return styles.premium;
            break;
    }
};

const getCopy = (adType) => {
    const standOutCopy = () => (
        <div>
            <Text critical>Company logo to promote your brand</Text>
            <Text critical>Additional 3 selling points to sell your role</Text>
            <Text critical>Bold, eye-catching border</Text>
        </div>
    );

    if (adType === STANDOUT) {
        return standOutCopy();
    }

    if (adType === PREMIUM) {
        return (
            <div>
                {standOutCopy()}
                <Text critical strong>Priority position at the top of search results for 7 days</Text>
            </div>
        )
    }
};

const AdTypeMarketing = ({ adType }) => {
    return (
        <div className={getClassName(adType)}>
            <Text>30 days on site allowing you to make unlimited changes</Text>
            <Text>Matched to relevant candidates via email</Text>
            <Text>Appears across mobile, desktop and tablet devices</Text>
            <Text>Access to proactively find profiles on SEEK Talent Search</Text>
            <Text>Free candidate management tools</Text>
            {getCopy(adType)}
        </div>
    )
};

export default AdTypeMarketing;

