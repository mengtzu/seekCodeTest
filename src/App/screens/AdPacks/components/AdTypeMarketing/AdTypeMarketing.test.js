import React from 'react';
import { AdTypeMarketing } from './AdTypeMarketing';
import renderer from 'react-test-renderer';

import { CLASSIC, STANDOUT, PREMIUM } from '../../../../../shared/constants/products';

it('renders correctly for Classic', () => {
    const tree = renderer.create(
        <AdTypeMarketing adType={CLASSIC} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly for Standout', () => {
    const tree = renderer.create(
        <AdTypeMarketing adType={STANDOUT} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly for PREMIUM', () => {
    const tree = renderer.create(
        <AdTypeMarketing adType={PREMIUM} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});