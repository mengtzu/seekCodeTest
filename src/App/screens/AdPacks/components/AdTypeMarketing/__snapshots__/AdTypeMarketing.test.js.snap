// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for Classic 1`] = `
<div
  className="classic"
>
  <div
    className="root"
  >
    <span
      className=""
    >
      30 days on site allowing you to make unlimited changes
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Matched to relevant candidates via email
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Appears across mobile, desktop and tablet devices
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Access to proactively find profiles on SEEK Talent Search
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Free candidate management tools
    </span>
  </div>
</div>
`;

exports[`renders correctly for PREMIUM 1`] = `
<div
  className="premium"
>
  <div
    className="root"
  >
    <span
      className=""
    >
      30 days on site allowing you to make unlimited changes
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Matched to relevant candidates via email
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Appears across mobile, desktop and tablet devices
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Access to proactively find profiles on SEEK Talent Search
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Free candidate management tools
    </span>
  </div>
  <div>
    <div>
      <div
        className="root"
      >
        <span
          className=""
        >
          Company logo to promote your brand
        </span>
      </div>
      <div
        className="root"
      >
        <span
          className=""
        >
          Additional 3 selling points to sell your role
        </span>
      </div>
      <div
        className="root"
      >
        <span
          className=""
        >
          Bold, eye-catching border
        </span>
      </div>
    </div>
    <div
      className="root"
    >
      <span
        className="root"
      >
        Priority position at the top of search results for 7 days
      </span>
    </div>
  </div>
</div>
`;

exports[`renders correctly for Standout 1`] = `
<div
  className="standout"
>
  <div
    className="root"
  >
    <span
      className=""
    >
      30 days on site allowing you to make unlimited changes
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Matched to relevant candidates via email
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Appears across mobile, desktop and tablet devices
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Access to proactively find profiles on SEEK Talent Search
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Free candidate management tools
    </span>
  </div>
  <div>
    <div
      className="root"
    >
      <span
        className=""
      >
        Company logo to promote your brand
      </span>
    </div>
    <div
      className="root"
    >
      <span
        className=""
      >
        Additional 3 selling points to sell your role
      </span>
    </div>
    <div
      className="root"
    >
      <span
        className=""
      >
        Bold, eye-catching border
      </span>
    </div>
  </div>
</div>
`;
